pytorch_lightning/utilities/warnings.py

```diff
@@ -41,13 +41,10 @@ class PossibleUserWarning(UserWarning):
     """Warnings that could be false positives."""
 
 
-class LightningDeprecationWarning(DeprecationWarning):
+class LightningDeprecationWarning(FutureWarning):
     """Deprecation warnings raised by PyTorch Lightning."""
 
 
-# enable our warnings
-warnings.simplefilter("default", category=LightningDeprecationWarning)
-
 rank_zero_deprecation = partial(rank_zero_warn, category=LightningDeprecationWarning)
```

Comment:
We can do this direclty and remove `LightningDeprecationWarning`

```suggestion
rank_zero_deprecation = partial(rank_zero_warn, category=FutureWarning)
```

Reply:
if we do this, will we be able to catch unrecognized deprecations inside tests?
https://github.com/PyTorchLightning/pytorch-lightning/blob/42fddc371af0c007b06d25c35059d84a116b3df0/setup.cfg#L27-L33

Reply:
If we don't have our original warning, is it possible to differentiate `FutureWarning` from PL and dependencies?

Reply:
:eyes: https://github.com/akihironitta/pytorch-lightning/pull/4

Reply:
Since we don't have failures in #11541, we can get rid of our custom class and use `FutureWarning` directly.

This also has the advantage of fixing the problem where pytest hides import errors because our custom warning class is not importable.